src/ignition/ignition.go

```diff
@@ -78,3 +83,32 @@ func (i *ignition) MergeIgnitionConfig(base *types.Config, overrides *types.Conf
 	}
 	return &config, nil
 }
+
+func (i *ignition) InjectKubeletTempPrivateKey(pathToSourceIgnition string, privateKeyBytes []byte, certPathToInject string) error {
+	sourceIgnition, err := i.ParseIgnitionFile(pathToSourceIgnition)
+	if err != nil {
+		return errors.Wrapf(err, "unable to parse ignition file %s", pathToSourceIgnition)
+	}
+	privateKeyFile := types.File{
+		Node: types.Node{
+			Overwrite: swag.Bool(true),
+			Path: certPathToInject,
+			User: types.NodeUser{
+				Name: swag.String("root"),
+			},
+		},
+		FileEmbedded1: types.FileEmbedded1{
+			Contents: types.Resource{
+				Source: swag.String(
+					fmt.Sprintf(
+						"data:text/plain;charset=utf-8;base64,%s",
+						base64.StdEncoding.EncodeToString([]byte(privateKeyBytes)),
+					),
+				),
+			},
+			Mode: swag.Int(420),
```

Comment:
I don't like bringing in swag as a dependency into this code, just to abuse its shortcut for making int pointers. Also use Octal literals:
```suggestion
			Mode: swag.Int(0644),
```

Also `644` is inappropriate for a private key, it should be `600`, we want the to restrict access to it as much as possible

![image](https://github.com/user-attachments/assets/e9d0ba09-7a52-416c-98d8-980d1566839d)

![image](https://github.com/user-attachments/assets/d4b9cfb1-ff56-4521-b725-b29b5f5ee025)